### PR TITLE
Geopackage flexibility, Oracle connections, maintenance

### LINKED ISSUE #74: use keyring instead of getOption for creds
I suggest moving councilR functions over to using `keyring` package for storing credentials as a more secure option than the `getOption()` in current usage, to set one best practices standard across Council repos/packages

### LINKED ISSUE #83: Replace database documentation link
Link to MT SI onboarding

### LINKED ISSUE #84: Add passthrough arguments in import_from_gpkg()
Some geopackages have multiple layers. We should be able to pass on which layer we want to `sf::read_sf()`

### LINKED ISSUE #87: FRED-Oracle connections
We still rely on our Oracle databases, and need to have connections using RJDBC

### LINKED ISSUE #88: Use internal type checkers
Right now, we are importing internal rlang functions, which is an unstable workflow.

Adapt from rlang

### PR DESCRIPTION
**Pull request by eroten**
- Dependent packages updated
- Updated custom scales parameters to comply with ggplot2 3.5.0
- `import_from_gpkg()` now takes `...` pass-through arguments. These are passed to `sf::read_sf()` #84 
- `import_from_gpkg()` is now flexible with the geopackage file name in cases where the geopackage name doesn't match the zipped/downloaded file name. #86 
- `import_from_gis()` now takes parameter `geometry` which determines whether to include geographic data in the returned table. This is useful for pulling tables that would otherwise be very large with the spatial data included. #78 
- New set of functions for checking parameter types. This replaces all references to internal rlang functions. Closes #88 
- Updated testing, documentation
- Added Oracle database connections, `fred_oracle_connection()` support closes #87 .
- All credential management is through `keyring`. Closes #74

### CODE REVIEW

File: R/db_emissions.R
Comment (Brandon-Whited) [inline R/db_emissions.R:1]: I got a "Database failed to connect" when running `emissions_connection()`. Not sure if it's a permissions issue, a dns issue, or something else.

File: tests/testthat/test-db_FRED.R [hunk 1]
```diff
@@ -27,6 +27,23 @@ testthat::test_that("upercase group quarters table", {
   testthat::expect_equal(class(gq)[[1]], "data.frame")
 })
 
+# testthat::test_that("test group quarters table", {
+#   gq <- import_from_FRED(
+#     table_name = "GQ_UNIT",
+#     prod = FALSE,
+#     uid = httr2::secret_decrypt("QUHBRb_yoy2RRj59qno8NVXA7mW402xkins", "COUNCILR_KEY"),
+#     pwd = httr2::secret_decrypt("tsGlVlM0KBdLcUPNLX5f0Hll2_6HFF_rQx62cw", "COUNCILR_KEY"),
```
Comment (Brandon-Whited) [inline tests/testthat/test-db_FRED.R:35]: I've not used `httr2::secret_decrypt`. Is there a reason that we wouldn't use keychain here too?

**Review by Brandon-Whited — APPROVED** [review]
I tested what I could and your code looks great!

File: tests/testthat/test-db_FRED.R (hunk 1 above)
Comment (eroten) [reply tests/testthat/test-db_FRED.R:35]: testthat environments are very picky - I've found that this works across machines and lets us use an environment variable in Github Actions for testing.

File: R/db_emissions.R
Comment (eroten) [reply R/db_emissions.R:1]: Its likely permissions on your end. Thanks for trying!

**Pull request merged**